File: typescript/tutor_agent/src/workflows.ts
/**
 * Tutor Workflows for Educational Interactions
 *
 * Demonstrates the agent handoff pattern where a triage agent routes
 * the student's question to the appropriate specialist tutor.
 */

import { workflow } from '@agnt5/sdk';
import type { Context } from '@agnt5/sdk';
import { getTutorAgent } from './agents.js';

export const tutorChatWorkflow = workflow(
  'tutor_chat_workflow',
  async (ctx: Context, input: { message: string }) => {
    const { message } = input;
    ctx.logger.info(`Tutor chat workflow - message: ${message}`);

    try {
      const result = await getTutorAgent().run(message, ctx);
      return {
        output: result.output,
      };
    } catch (err) {
      const msg = err instanceof Error ? err.message : String(err);
      ctx.logger.error(`Error running tutor agent: ${msg}`);
      return {
        output: `I apologize, but I'm having trouble processing your question right now. Could you please rephrase: ${message}`,
      };
    }
  },
);

File: go/tutor_agent/workflows.go
// Tutor workflow for educational interactions.
//
// Demonstrates the agent handoff pattern: a triage agent delegates to
// specialized tutors (see agents.go) based on the subject of the question.
package main

import "agnt5.dev/sdk-go/agnt5"

type TutorChatInput struct {
	Message string `json:"message"`
}

type TutorChatOutput struct {
	Output string `json:"output"`
}

func tutorChatWorkflow(ctx *agnt5.Context, in TutorChatInput) (TutorChatOutput, error) {
	ctx.Logger().Info("Tutor chat workflow", "message", in.Message)

	result, err := triageAgent.Run(ctx, agnt5.AgentInput{Message: in.Message})
	if err != nil {
		ctx.Logger().Error("Error running tutor agent", "error", err)
		return TutorChatOutput{
			Output: "I apologize, but I'm having trouble processing your question right now. Could you please rephrase: " + in.Message,
		}, nil
	}

	return TutorChatOutput{Output: result.Response}, nil
}
